recipes/libcurl/all/conanfile.py

```diff
@@ -596,7 +545,14 @@ def _generate_with_cmake(self):
         tc.variables["CURL_ZLIB"] = self.options.with_zlib
         tc.variables["CURL_BROTLI"] = self.options.with_brotli
         tc.variables["CURL_ZSTD"] = self.options.with_zstd
+        if self.options.with_zstd:
+            # Can't force CMakeDeps to generate this
+            tc.cache_variables["ZSTD_FOUND"] = True
```

Comment:
This is the main stopping point for this PR. Ideally we would have a way that CMakeDeps listens to the set_properties for these versions

Reply:
Discussion in https://github.com/conan-io/conan/issues/17483